test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -0,0 +1,170 @@
+  return {size.rows(), size.cols()};
+}
+
+const std::vector<std::tuple<SizeType, SizeType, SizeType, SizeType>> sizes =
+    // m, n, mb, nb
+    {{3, 0, 1, 1},   {0, 5, 2, 3},  // m, n = 0
+     {2, 2, 3, 3},   {3, 4, 6, 7},  // m < mb
+     {3, 3, 1, 1},   {4, 4, 2, 2}, {6, 3, 3, 3}, {12, 2, 4, 4}, {12, 24, 3, 3},
+     {24, 36, 6, 6}, {5, 8, 3, 2}, {4, 6, 2, 3}, {5, 5, 2, 3},  {8, 27, 3, 4},
+     {15, 34, 4, 6}, {7, 6, 2, 3}, {8, 5, 2, 3}, {7, 5, 2, 3}};
+
+template <class T>
+MatrixLocal<T> makeLocal(const Matrix<const T, Device::CPU>& matrix) {
+  return {matrix.size(), matrix.distribution().blockSize()};
+}
+
+template <class T>
+void set_zero(Matrix<T, Device::CPU>& mat) {
+  dlaf::matrix::util::set(mat, [](auto&&) { return static_cast<T>(0.0); });
+}
+
+template <class T>
+void getTau(T& tau, T dotprod, BaseType<T> /*tau_i*/) {
+  tau = static_cast<T>(2.0) / dotprod;
+}
+
+template <class T>
+void getTau(std::complex<T>& tau, T dotprod, BaseType<T> tau_i) {
+  tau = {(static_cast<T>(1.0) + sqrt(static_cast<T>(1.0) - dotprod * tau_i * tau_i)) / dotprod, tau_i};
+}
+
+template <class T>
+void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType nb) {
+  LocalElementSize sizeC(m, n);
+  TileElementSize blockSizeC(mb, nb);
+  Matrix<T, Device::CPU> mat_c(sizeC, blockSizeC);
+  dlaf::matrix::util::set_random(mat_c);
+
+  LocalElementSize sizeV(m, m);
+  TileElementSize blockSizeV(mb, mb);
+  Matrix<T, Device::CPU> mat_v(sizeV, blockSizeV);
+  dlaf::matrix::util::set_random(mat_v);
+
+  SizeType tottaus;
+  if (m < mb || m == 0 || n == 0)
+    tottaus = 0;
+  else
+    tottaus = m - mb;
+
+  if (tottaus > 0) {
+    // Reset diagonal and upper values of V
+    MatrixLocal<T> v({m, m}, blockSizeV);
+    for (const auto& ij_tile : iterate_range2d(v.nrTiles())) {
+      // copy only the panel
+      const auto& source_tile = mat_v.read(ij_tile).get();
+      if (ij_tile.row() <= ij_tile.col()) {
+        tile::set0<T>(v.tile(ij_tile));
+      }
+      else if (ij_tile.row() == ij_tile.col() + 1) {
+        copy(source_tile, v.tile(ij_tile));
+        tile::laset<T>(lapack::MatrixType::Upper, 0.f, 1.f, v.tile(ij_tile));
+      }
+      else {
+        copy(source_tile, v.tile(ij_tile));
+      }
+    }
+
+    // Create C local
+    MatrixLocal<T> c({m, n}, blockSizeC);
+    for (const auto& ij_tile : iterate_range2d(c.nrTiles())) {
+      // copy only the panel
+      const auto& source_tile = mat_c.read(ij_tile).get();
+      copy(source_tile, c.tile(ij_tile));
+    }
+
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus;
+
+    common::internal::vector<T> tausloc;
+
+    // Impose orthogonality: Q = I - v tau vH is orthogonal (Q QH = I).
+    // Real case: tau = 2 / (vH v)
+    // Complex case: real part of tau = [1 + sqrt(1 - vH v taui^2)]/(vH v)
+    for (SizeType k = 0; k < tottaus; k += mb) {
+      common::internal::vector<T> t_tile;

```

Comment:
Maybe it is better to call it `tau_tile`.

Use reserve to allocate mb elements.